spec/src/antlr/JDQL.g4

```diff
@@ -0,0 +1,92 @@
+grammar JDQL;
+
+statement : select_statement | update_statement | delete_statement;
+
+select_statement : select_clause? from_clause? where_clause? orderby_clause?;
+update_statement : 'UPDATE' entity_name set_clause where_clause?;
+delete_statement : 'DELETE' from_clause where_clause?;
+
+from_clause : 'FROM' entity_name;
+
+where_clause : 'WHERE' conditional_expression;
+
+set_clause : 'SET' update_item (',' update_item)*;
+update_item : state_field_path_expression '=' (scalar_expression | 'NULL');
+
+select_clause : 'SELECT' select_list;
+select_list
+    : scalar_expression (',' scalar_expression)* // could limit to single select item and/or to state_field_path_expression
+    | aggregate_expression
+    ;
+aggregate_expression : 'COUNT' '(' '*' ')';
+
+orderby_clause : 'ORDER' 'BY' orderby_item (',' orderby_item)*;
+orderby_item : state_field_path_expression ('ASC' | 'DESC');
+
+conditional_expression
+    // highest to lowest precedence
+    : '(' conditional_expression ')'
+    | null_comparison_expression
+    | in_expression
+    | between_expression
+    | like_expression
+    | comparison_expression
+    | 'NOT' conditional_expression
+    | conditional_expression 'AND' conditional_expression
+    | conditional_expression 'OR' conditional_expression
+    ;
+
+comparison_expression : scalar_expression ('=' | '>' | '>=' | '<' | '<=' | '<>') scalar_expression;
+between_expression : scalar_expression 'NOT'? 'BETWEEN' scalar_expression 'AND' scalar_expression;
+like_expression : scalar_expression 'NOT'? 'LIKE' STRING;
+
+in_expression : state_field_path_expression 'NOT'? 'IN' '(' in_item (',' in_item)* ')';
+in_item : literal | enum_literal | input_parameter; // could simplify to just literal
+
+null_comparison_expression : state_field_path_expression 'IS' 'NOT'? 'NULL';
+
+scalar_expression
+    // highest to lowest precedence
+    : '(' scalar_expression ')'
+    | primary_expression
+    | ('+' | '-') scalar_expression
+    | scalar_expression ('*' | '/') scalar_expression
+    | scalar_expression ('+' | '-') scalar_expression
+    | scalar_expression '||' scalar_expression
+    ;
+
+primary_expression
+    : function_expression
+    | special_expression
+    | state_field_path_expression
+    | enum_literal
+    | input_parameter
+    | literal
+    ;
+
+function_expression
+    : 'ABS' '(' scalar_expression ')'
+    | 'LENGTH' '(' scalar_expression ')'
+    | 'LOWER' '(' scalar_expression ')'
+    | 'UPPER' '(' scalar_expression ')'
+    | 'LEFT' '(' scalar_expression ',' scalar_expression ')'
+    | 'RIGHT' '(' scalar_expression ',' scalar_expression ')'
+    ;
+
+special_expression
+    : 'LOCAL' 'DATE'
+    | 'LOCAL' 'DATETIME'
+    | 'TRUE'
+    | 'FALSE'
+    ;
+
+state_field_path_expression : IDENTIFIER ('.' IDENTIFIER)*;
```

Comment:
> We will need to keep track of whether the issue/pull to make identification variables optional made/makes it into Jakarta Persistence 3.2. Hopefully it will. If it doesn't, we would need to require them as well and make them optional in the future.

I just got the go ahead from @lukasj. We are going to make this change to JPQL.

Reply:
> I just got the go ahead from @lukasj. We are going to make this change to JPQL.

Excellent! This will enable Jakarta Data's Query Language to omit identification variables and still be a subset of JPQL, which clears up the main concern I had with the query language.